display_list/display_list_builder.cc

```diff
@@ -57,13 +57,16 @@ sk_sp<DisplayList> DisplayListBuilder::Build() {
   nested_bytes_ = nested_op_count_ = 0;
   storage_.realloc(bytes);
   bool compatible = layer_stack_.back().is_group_opacity_compatible();
-  return sk_sp<DisplayList>(new DisplayList(storage_.release(), bytes, count,
-                                            nested_bytes, nested_count,
-                                            cull_rect_, compatible));
+  return sk_sp<DisplayList>(new DisplayList(
+      storage_.release(), bytes, count, nested_bytes, nested_count, cull_rect_,
+      compatible, complexity_accumulator_->complexity_score(),
+      complexity_accumulator_->should_be_cached()));
 }
 
 DisplayListBuilder::DisplayListBuilder(const SkRect& cull_rect)
-    : cull_rect_(cull_rect) {
+    : cull_rect_(cull_rect), complexity_accumulator_(nullptr) {
+  complexity_accumulator_ =
+      std::make_unique<DisplayListNaiveComplexityCalculator>();
```

Comment:
Where will this come from in the end? Will we have an idea of which backend complexity calculator to use when DLs are created? Should this be a parameter on the constructor (with a default of null or the Base fallback calculator?)